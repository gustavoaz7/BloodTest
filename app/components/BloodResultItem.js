import React from 'react';
import { View } from 'react-native';
import PropTypes from 'prop-types';
// eslint-disable-next-line import/no-extraneous-dependencies
import { Feather } from '@expo/vector-icons';
import styled, { withTheme } from 'styled-components/native';
import Text from './base/Text';
import Card from './base/Card';
import { resultType, themeType } from '../types';
import Touchable from './base/Touchable';

const ICON_MAP = {
  morning: 'sunrise',
  afternoon: 'sunset',
  evening: 'moon',
};

function BloodResultItem({ onPress, item: { name, email, samples }, theme }) {
  return (
    <Touchable onPress={onPress}>
      <Card>
        <Header>
          <Feather name="user" size={40} color={theme.primary} />
          <Profile>
            <Name>{name}</Name>
            <Email numberOfLines={1}>{email}</Email>
          </Profile>
        </Header>
        <Samples>
          {samples.map(({ time, value }) => (
            <SampleBox key={`${time}-${value}`}>
              <Feather name={ICON_MAP[time]} size={25} color={theme.secondary} />
              <SampleValue>{value}</SampleValue>
            </SampleBox>
          ))}
        </Samples>
      </Card>
    </Touchable>
  );
}

const Row = styled(View)`
  flex-direction: row;
`;

const Header = styled(Row)`
  margin-bottom: 4px;
`;

const Profile = styled(View)`
  flex: 1;
  justify-content: space-around;
  margin-left: 10px;
`;

export const Name = styled(Text)`
  font-weight: bold;
`;

export const Email = styled(Text)`
  color: 'rgba(0, 0, 0, 0.5)';
`;

export const Samples = styled(Row)`
  justify-content: space-around;
  margin: 8px 0;
`;

const SampleBox = styled(View)`
  align-items: center;
`;

export const SampleValue = styled(Text)`
  margin-top: 4px;
`;

BloodResultItem.propTypes = {
  item: resultType.isRequired,
  theme: themeType.isRequired,
  onPress: PropTypes.func,
};

BloodResultItem.defaultProps = {
  onPress() {},
};

export default withTheme(BloodResultItem);
