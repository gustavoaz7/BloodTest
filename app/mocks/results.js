export const resultsMock = [
  {
    id: '0b82fed2-a091-4318-ba02-4d5e6192d6f7',
    name: 'Citlalli Kuhic',
    email: 'Berniece.Haley42@hotmail.com',
    samples: [
      {
        time: 'morning',
        value: 0.48,
      },
      {
        time: 'afternoon',
        value: 0.39,
      },
      {
        time: 'evening',
        value: 0.19,
      },
    ],
  },
  {
    id: 'acf65e6a-1878-4cd9-bf6f-440dafd4fa18',
    name: 'Rachael Lehner',
    email: 'Teagan_Ratke83@gmail.com',
    samples: [
      {
        time: 'morning',
        value: 0.67,
      },
      {
        time: 'afternoon',
        value: 0.31,
      },
      {
        time: 'evening',
        value: 0.24,
      },
    ],
  },
  {
    id: 'ca285be1-0d6b-42f6-bfec-b3aeaa2aaf8f',
    name: 'Willa Quitzon',
    email: 'Palma17@yahoo.com',
    samples: [
      {
        time: 'morning',
        value: 0.73,
      },
      {
        time: 'afternoon',
        value: 0.19,
      },
      {
        time: 'evening',
        value: 0.14,
      },
    ],
  },
  {
    id: '95eba822-a7b3-4938-8c94-005733701c1f',
    name: 'Miss Audie Moore',
    email: 'Giuseppe19@hotmail.com',
    samples: [
      {
        time: 'morning',
        value: 0.26,
      },
      {
        time: 'afternoon',
        value: 0.09,
      },
      {
        time: 'evening',
        value: 0.29,
      },
    ],
  },
  {
    id: '4fea89aa-bcf7-42b0-a5e8-202fc03a02fe',
    name: 'Yesenia Nicolas',
    email: 'Maeve_Jacobs14@yahoo.com',
    samples: [
      {
        time: 'morning',
        value: 0.78,
      },
      {
        time: 'afternoon',
        value: 0.26,
      },
      {
        time: 'evening',
        value: 0.24,
      },
    ],
  },
  {
    id: '17d11f05-2580-49b5-bb17-b786209c4792',
    name: 'Ms. Harmon Kunze',
    email: 'Johnson90@yahoo.com',
    samples: [
      {
        time: 'morning',
        value: 0.47,
      },
      {
        time: 'afternoon',
        value: 0.09,
      },
      {
        time: 'evening',
        value: 0.09,
      },
    ],
  },
];
