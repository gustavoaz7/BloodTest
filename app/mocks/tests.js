export const testsMock = [
  {
    id: '5d41b693ed44f7ee9f9436d0',
    title: 'Healthcare',
    date: '17.07.2019',
    result: [
      {
        biomarker: 'fats',
        color: '#e30613',
        percentage: 25,
      },
      {
        biomarker: 'carbohydrates',
        color: '#D3235D',
        percentage: 25,
      },
      {
        biomarker: 'vitamins',
        color: '#547F81',
        percentage: 25,
      },
      {
        biomarker: 'magnesium',
        color: '#0085FF',
        percentage: 25,
      },
    ],
  },
  {
    id: '5d41b6938f1360cdde5ecc89',
    title: 'Advanced',
    date: '08.07.2019',
    result: [
      {
        biomarker: 'protein',
        color: '#E2D669',
        percentage: 25,
      },
      {
        biomarker: 'vitamins',
        color: '#ffcc00',
        percentage: 25,
      },
      {
        biomarker: 'magnesium',
        color: '#e30613',
        percentage: 25,
      },
      {
        biomarker: 'carbohydrates',
        color: '#0085FF',
        percentage: 25,
      },
    ],
  },
  {
    id: '5d41b693084f4f55217fd864',
    title: 'Sport',
    date: '30.07.2019',
    result: [
      {
        biomarker: 'fiber',
        color: '#E2D669',
        percentage: 25,
      },
      {
        biomarker: 'fats',
        color: '#D3235D',
        percentage: 25,
      },
      {
        biomarker: 'magnesium',
        color: '#547F81',
        percentage: 25,
      },
      {
        biomarker: 'vitamins',
        color: '#0085FF',
        percentage: 25,
      },
    ],
  },
];
