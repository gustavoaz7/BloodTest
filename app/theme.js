export const theme = {
  primary: '#f5785a',
  secondary: '#321b3c',
  fontFamily: 'PostGrotesk-Light',
};
