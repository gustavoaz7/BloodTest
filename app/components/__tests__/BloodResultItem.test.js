import React from 'react';
import { shallow } from 'enzyme';
import { ThemeProvider } from 'styled-components/native';
// eslint-disable-next-line import/no-extraneous-dependencies
import { Feather } from '@expo/vector-icons';
import BloodResultItem, { Samples, Name, Email, SampleValue } from '../BloodResultItem';
import { resultsMock } from '../../mocks/results';
import { theme } from '../../theme';

// eslint-disable-next-line react/prop-types
function Wrapper({ children }) {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
}

describe('BloodResultItem', () => {
  const result = resultsMock[0];
  const options = { wrappingComponent: Wrapper };

  it('renders without crashing', () => {
    shallow(<BloodResultItem item={result} />, options);
  });

  it('renders profile correctly', () => {
    const wrapper = shallow(<BloodResultItem item={result} />, options)
      .dive()
      .dive();

    expect(wrapper.find(Name).text()).toBe(result.name);
    expect(wrapper.find(Email).text()).toBe(result.email);
  });

  it('renders provided samples correctly', () => {
    const item = {
      ...result,
      samples: [
        {
          time: 'morning',
          value: 0.51,
        },
        {
          time: 'afternoon',
          value: 0.36,
        },
        {
          time: 'evening',
          value: 0.14,
        },
      ],
    };
    const wrapper = shallow(<BloodResultItem item={item} />, options)
      .dive()
      .dive();

    const samples = wrapper.find(Samples);
    expect(samples.children()).toHaveLength(item.samples.length);

    const icons = samples.find(Feather);
    expect(icons.at(0).prop('name')).toBe('sunrise');
    expect(icons.at(1).prop('name')).toBe('sunset');
    expect(icons.at(2).prop('name')).toBe('moon');

    wrapper
      .find(SampleValue)
      .forEach((value, index) => expect(value.text()).toEqual(`${item.samples[index].value}`));
  });

  it('matches snapshot', () => {
    const wrapper = shallow(<BloodResultItem item={result} />, options)
      .dive()
      .dive();
    expect(wrapper).toMatchSnapshot();
  });
});
