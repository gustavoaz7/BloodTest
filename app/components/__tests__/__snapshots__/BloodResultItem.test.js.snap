// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BloodResultItem matches snapshot 1`] = `
<Card
  style={null}
>
  <Styled(Styled(View))>
    <Icon
      allowFontScaling={false}
      color="#f5785a"
      name="user"
      size={40}
    />
    <Styled(View)>
      <Styled(Styled(Text))>
        Citlalli Kuhic
      </Styled(Styled(Text))>
      <Styled(Styled(Text))
        numberOfLines={1}
      >
        Berniece.Haley42@hotmail.com
      </Styled(Styled(Text))>
    </Styled(View)>
  </Styled(Styled(View))>
  <Styled(Styled(View))>
    <Styled(View)
      key="morning-0.48"
    >
      <Icon
        allowFontScaling={false}
        color="#321b3c"
        name="sunrise"
        size={25}
      />
      <Styled(Styled(Text))>
        0.48
      </Styled(Styled(Text))>
    </Styled(View)>
    <Styled(View)
      key="afternoon-0.39"
    >
      <Icon
        allowFontScaling={false}
        color="#321b3c"
        name="sunset"
        size={25}
      />
      <Styled(Styled(Text))>
        0.39
      </Styled(Styled(Text))>
    </Styled(View)>
    <Styled(View)
      key="evening-0.19"
    >
      <Icon
        allowFontScaling={false}
        color="#321b3c"
        name="moon"
        size={25}
      />
      <Styled(Styled(Text))>
        0.19
      </Styled(Styled(Text))>
    </Styled(View)>
  </Styled(Styled(View))>
</Card>
`;
